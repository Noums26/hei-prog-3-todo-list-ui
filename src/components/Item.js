import React from "react";

const Item = ({ todo, position, changeTodoState }) => {
  const toDone = () => {
    let temp = { ...todo, state: true };
    changeTodoState(position, temp);
  };
  return (
    <div className="item">
      {!todo.state && <input type="checkbox" onChange={toDone} />}
      <span>{todo.message}</span>
    </div>
  );
};

export default Item;
