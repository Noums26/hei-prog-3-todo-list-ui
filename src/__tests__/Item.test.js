import { create } from "react-test-renderer";
import Item from "../components/Item";

test("should test todo snapshot", () => {
  const todo = { message: "Tepr 21", state: false };
  const item = create(
    <Item todo={todo} position={1} changeTodoState={jest.fn()} />
  ).toJSON();

  expect(item).toMatchSnapshot();
});

test("should test done snapshot", () => {
  const todo = { message: "Tepr 21", state: true };
  const item = create(
    <Item todo={todo} position={1} changeTodoState={jest.fn()} />
  ).toJSON();

  expect(item).toMatchSnapshot();
});
