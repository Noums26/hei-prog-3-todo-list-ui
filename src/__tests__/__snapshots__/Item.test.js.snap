// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test done snapshot 1`] = `
<div
  className="item"
>
  <span>
    Tepr 21
  </span>
</div>
`;

exports[`should test todo snapshot 1`] = `
<div
  className="item"
>
  <input
    onChange={[Function]}
    type="checkbox"
  />
  <span>
    Tepr 21
  </span>
</div>
`;
